aem-authoring-toolkit-plugin/src/main/java/com/exadel/aem/toolkit/core/util/writer/ContentXmlWriter.java

```diff
@@ -67,10 +71,15 @@ boolean isProcessed(Class<?> componentClass) {
      */
     @Override
     void populateDomDocument(Class<?> componentClass, Element root) {
-        Dialog dialog = componentClass.getDeclaredAnnotation(Dialog.class);
+        Annotation annotation = componentClass.getDeclaredAnnotation(Component.class);
+        if (annotation == null) {
+            annotation = componentClass.getDeclaredAnnotation(Dialog.class);
+        }
         root.setAttribute(DialogConstants.PN_PRIMARY_TYPE, DialogConstants.NT_COMPONENT);
-        PluginRuntime.context().getXmlUtility().mapProperties(root, dialog, XmlScope.COMPONENT);
-        if(dialog.isContainer()) root.setAttribute(DialogConstants.PN_IS_CONTAINER, String.valueOf(true));
+        PluginRuntime.context().getXmlUtility().mapProperties(root, annotation, XmlScope.COMPONENT);
+        if (annotation instanceof Dialog && ((Dialog) annotation).isContainer()) {
```

Comment:
`@Component` can also have `isContainer` (something like the `getLayout()` method in the class below is needed)